### PR TITLE
Feature/SOF-6588 feat: add ConsistencyChecks schema

### PR DESCRIPTION
**Pull request by VsevolodX**

### CODE REVIEW

File: schema/system/consistency_checks.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "messages": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
```
Comment (timurbazhirov) [inline schema/system/consistency_checks.json]: Isolate to `consistency_check.json`

File: schema/system/consistency_checks.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    "key": {
+                        "type": "string"
+                    },
+                    "checkName": {
```
Comment (timurbazhirov) [inline schema/system/consistency_checks.json]: name

File: schema/in_memory_entity/has_consistency_checks.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
+    "$id": "in-memory-entity/has-consistency-checks",
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "Has consistency checks",
+    "allOf": [
+        {
+            "$ref": "base.json"
+        },
+        {
+            "$ref": "../system/consistency_checks.json"
```
Comment (timurbazhirov) [inline schema/in_memory_entity/has_consistency_checks.json]: Let's just add into the in-memory entity itself as optional key

File: schema/system/consistency_checks.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
+{
+    "$id": "system/consistency-checks",
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "consistency checks",
+    "properties": {
+        "keys": {
+            "type": "array",
```
Comment (timurbazhirov) [inline schema/system/consistency_checks.json]: We need to describe what these are

File: schema/in_memory_entity/has_consistency_checks.json (hunk 3 above)
Comment (VsevolodX) [reply schema/in_memory_entity/has_consistency_checks.json]: That means to the `in_memory_entity/base.json`

File: schema/in_memory_entity/base.json [hunk 5]
```diff
@@ -19,6 +19,12 @@
         },
         "systemName": {
             "type": "string"
+        },
+        "consistency_checks": {
```
Comment (timurbazhirov) [inline schema/in_memory_entity/base.json]: should be `camelCase`

File: schema/system/consistency_checks.json (hunk 4 above)
Comment (VsevolodX) [reply schema/system/consistency_checks.json]: Might look cumbersome, but outlines the idea: "The output of consistency checks performed on data adhering to JSON schema, but inconsistent with scientific or logical rules, to show problems in UI."

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**